crates/stages/src/stages/index_storage_history.rs

```diff
@@ -80,12 +88,39 @@ impl<DB: Database> Stage<DB> for IndexStorageHistoryStage {
             return Ok(ExecOutput::done(input.checkpoint()))
         }
 
-        let (range, is_final_range) = input.next_block_range_with_threshold(self.commit_threshold);
+        let mut range = input.next_block_range();
+        let first_sync = input.checkpoint().block_number == 0;
 
-        let indices = provider.changed_storages_and_blocks_with_range(range.clone())?;
-        provider.insert_storage_history_index(indices)?;
+        // On first sync it's faster to rebuild the history from scratch.
+        if first_sync {
+            provider.tx_ref().clear::<tables::StoragesHistory>()?;
+            range = 0..=*input.next_block_range().end();
```

Comment:
isn't it already `0..=end`?

Reply:
No, it starts at block 1. But since before we clear the table (to allow `tx.append`) we need to do it from block 0 so it also iterates its changesets (on etherem mainnet, only accountshistory eg.)

Reply:
ah ok, genesis block. Can we add it in a comment?